include/envoy/network/io_handle.h

```diff
@@ -26,8 +24,11 @@ class IoHandle {
   /**
    * Clean up IoHandle resources
    */
-  virtual void close() PURE;
+  virtual Api::SysCallIntResult close() PURE;
+
+  virtual bool isClosed() const PURE;
```

Comment:
Also super nitty nit, but what do we think of making this isOpen() instead?  If we have an fd which is complete initialized, it feels a bit weird so say it is closed.

I'm totally open to leaving this as-is, just figured I'd call it out and see if anyone else agreed.

Reply:
makes sense. done